_source/_posts/2022-05-16-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,732 @@
+        registration:
+          oidc:
+            client-id: 0oa6ycm987987uy98
+            client-secret: "{cipher}88acb434dd088acb434dd088acb434dd0..."
+```
+
+Stop the app if you have not already using `control-c`. Restart it.
+
+```bash
+docker-compose up
+```
+
+Make sure you can still sign into the gateway: http://localhost:8080/
+
+You may be wondering (like I did initially), why can't I just put the client secret in the `.env` file? This doesn't work because the `.env` file is processed by Spring Cloud Config and JHipster registry after the container is composed, not Docker Compose during the container creation, which is when the `.env` file is processed.
+
+You're done with the Docker Compose implementation. To clean up, you can run the following command. This will stop and remove the containers, networks, volumes, and images created by `docker-compose up`.
+
+```bash
+docker-compose down --remove-orphans
+```
+
+## Create the Azure Kubernetes cluster
+
+The app works locally. Now it's time to deploy it to an Azure Kubernetes Cluster (AKS). The first step is to create an AKS cluster.
+
+It's super easy to use the CLI to create a cluster. I'll show you the command below. However, there's a wrinkle. The free tier cannot create a cluster in many of the regions because of resource quotas. At least, this was the case when I was working on this tutorial. Nor is there an easy way to quickly see what regions will allow you to create a free-tier cluster. This is why I used Australia East as the region--it allowed me to create a free cluster. 
+
+If the command below does not work, I suggest going to the Auzure portal and create a Kubernetes cluster there. Select **Create a service** and **Kubernetes Service**. You'll have to select different regions and see what sizes are available (under **Node size** and **Change size**) until you find a region that will allow you to create something in the free tier. But hopefully the command will work and you won't have to worry about it.
+
+The size I'm using for this tutorial is `Standard B4ms` with two nodes. I found that I needed two nodes for the cluster to start properly.
+
+Run the following command to create the AKS cluster.
+
+```bash
+az aks create --resource-group australia-east --name jhipster-demo --node-count 2 --enable-addons monitoring --generate-ssh-keys --node-vm-size standard_b4ms
+```
+
+This will probably take a few minutes.
+
+As a side note, at any point, you can stop the cluster. This will pause billing on the cluster.
+
+```bash
+az aks stop --resource-group australia-east --name jhipster-demo 
+```
+
+And you can start it again.
+
+```bash
+az aks start --resource-group australia-east --name jhipster-demo 
+```
+
+You can also stop and start the cluster from the Azure portal as well. 
+
+The next step is to get the credentials for the cluster and merge them into `.kube/confg` so that `kubectl` can use them. Use the following command.
```

Comment:
```suggestion
The next step is to get the credentials for the cluster and merge them into `.kube/config` so that `kubectl` can use them. Use the following command.
```